pkg/embeddedcluster/util.go

```diff
@@ -75,14 +87,17 @@ func GetCurrentInstallation(ctx context.Context) (*embeddedclusterv1beta1.Instal
 	if err != nil {
 		return nil, fmt.Errorf("failed to list installations: %w", err)
 	}
-	if len(installationList.Items) == 0 {
+	return installationList.Items, nil
+}
+
+func GetLatestInstallation(ctx context.Context, installations []embeddedclusterv1beta1.Installation) (*embeddedclusterv1beta1.Installation, error) {
```

Comment:
IMO this function name is confusing because there's already a `GetCurrentInstallation`. Listing installations from the cluster isn't an expensive process, so I would suggest basically calling both `ListInstallations` and `GetCurrentInstallation` instead of trying to save a function call.